### PR TITLE
Fix sle15 literal import

### PR DESCRIPTION
**Pull request by dcermak**

### CODE REVIEW

File: dist/package/openSUSE-release-tools.spec [hunk 1]
```diff
@@ -58,6 +58,10 @@ Requires:       python3-pycurl
 Requires:       python3-python-dateutil
 Requires:       python3-pyxdg
 Requires:       python3-requests
+# typing extensions are needed on SLE & Leap
+%if 0%{?suse_version} < 1600
```
Comment (Vogtinator) [inline dist/package/openSUSE-release-tools.spec:62]: Is it in ALP? If yes, can be changed to `<= 1500`

Comment (dcermak) [reply dist/package/openSUSE-release-tools.spec:62]: I hope that alp is at python >= 3.8! Then it wouldn't be required.

Comment (Vogtinator) [reply dist/package/openSUSE-release-tools.spec:62]: Should be 3.11

Comment (dcermak) [reply dist/package/openSUSE-release-tools.spec]: Fixed

File: .github/workflows/ci-test.yml [hunk 2]
```diff
@@ -42,6 +42,47 @@ jobs:
...
+
+      - name: fix the file permissions of the repository
+        run: chown -R $(id -un):$(id -gn) .
+
+      - name: add the openSUSE:Tools repository for Tumbleweed
+        run: zypper -n ar https://download.opensuse.org/repositories/openSUSE:/Tools/openSUSE_Tumbleweed/openSUSE:Tools.repo
+        if: ${{ contains(matrix.image, 'tumbleweed') }}
+      - name: add the openSUSE:Tools repository for Leap
+        run: zypper -n ar https://download.opensuse.org/repositories/openSUSE:/Tools/15.5/openSUSE:Tools.repo
+        if: ${{ contains(matrix.image, 'leap') }}
```
Comment (Vogtinator) [inline .github/workflows/ci-test.yml:69]:
```suggestion
        if: ${{ contains(matrix.image, '15.5') }}
```

File: .github/workflows/ci-test.yml [hunk 3]
```diff
@@ -42,6 +42,47 @@ jobs:
...
+        run: zypper -n in git
+
+      - uses: actions/checkout@v3
+
+      - name: fix the file permissions of the repository
+        run: chown -R $(id -un):$(id -gn) .
+
+      - name: add the openSUSE:Tools repository for Tumbleweed
+        run: zypper -n ar https://download.opensuse.org/repositories/openSUSE:/Tools/openSUSE_Tumbleweed/openSUSE:Tools.repo
+        if: ${{ contains(matrix.image, 'tumbleweed') }}
```
Comment (Vogtinator) [inline .github/workflows/ci-test.yml:66]: IMO `if` should be above `run`

File: .github/workflows/ci-test.yml [hunk 4]
```diff
@@ -42,6 +42,47 @@ jobs:
...
+        if: ${{ contains(matrix.image, 'leap') }}
+
+      - name: install the dependencies of openSUSE-release-tool
+        run: |
+          zypper -n --gpg-auto-import-keys refresh
+          zypper -n source-install openSUSE-release-tools
+
+      - name: FIXME, install missing dependencies
+        run: |
+          zypper -n in python3-requests python3-typing_extensions
```
Comment (Vogtinator) [inline .github/workflows/ci-test.yml:78]: The latter only on 15.5?

File: .github/workflows/ci-test.yml [hunk 5]
```diff
@@ -42,6 +42,47 @@ jobs:
...
+
+      - name: run a simple smoke test whether --help actually works
+        run: |
+          for f in $(find . -maxdepth 1 -type f -executable -print); do
+              eval "$f --help"
```
Comment (Vogtinator) [inline .github/workflows/ci-test.yml:83]:
```suggestion
              "$f" --help
```

File: .github/workflows/ci-test.yml (hunk 3 above)
Comment (dcermak) [reply .github/workflows/ci-test.yml:66]: It doesn't matter where you put the `if`.

File: .github/workflows/ci-test.yml (hunk 4 above)
Comment (dcermak) [reply .github/workflows/ci-test.yml:78]: requests is missing on both os versions

File: .github/workflows/ci-test.yml (hunk 3 above)
Comment (Vogtinator) [reply .github/workflows/ci-test.yml:66]: Not semantically, but usually conditions are written before the code that they guard, to match the flow of reading and thought.

File: .github/workflows/ci-test.yml [hunk 6]
```diff
@@ -42,6 +42,47 @@ jobs:
...
+        run: zypper -n ar https://download.opensuse.org/repositories/openSUSE:/Tools/openSUSE_Tumbleweed/openSUSE:Tools.repo
+        if: ${{ contains(matrix.image, 'tumbleweed') }}
+      - name: add the openSUSE:Tools repository for Leap
+        run: zypper -n ar https://download.opensuse.org/repositories/openSUSE:/Tools/15.5/openSUSE:Tools.repo
+        if: ${{ contains(matrix.image, 'leap') }}
+
+      - name: install the dependencies of openSUSE-release-tool
+        run: |
+          zypper -n --gpg-auto-import-keys refresh
+          zypper -n source-install openSUSE-release-tools
```
Comment (Vogtinator) [inline .github/workflows/ci-test.yml:74]:
```suggestion
          zypper -n source-install -d openSUSE-release-tools
```

for dependencies only.

But don't we actually need *runtime* requires here?

File: .github/workflows/ci-test.yml (hunk 3 above)
Comment (dcermak) [reply .github/workflows/ci-test.yml:65]: :shrug: done

File: .github/workflows/ci-test.yml (hunk 6 above)
Comment (dcermak) [reply .github/workflows/ci-test.yml:75]: We do need them, that's why requests was missing.

Comment (Vogtinator) [reply .github/workflows/ci-test.yml:75]: So `zypper in openSUSE-release-tools` would be more correct, though one "generation" behind?

Comment (dcermak) [reply .github/workflows/ci-test.yml:75]: yes, that's now in the job as well

**Comment by dcermak** [comment]
The test failure is a failure uploading to codecov.

**Review by Vogtinator — APPROVED** [review]

**Pull request merged**